src/resolve/resolved-dns-cache.c

```diff
@@ -547,6 +547,20 @@ static int dns_cache_put_positive(
         TAKE_PTR(i);
         return 0;
 }
+/* https://www.iana.org/assignments/special-use-domain-names/special-use-domain-names.xhtml */
+/* https://www.iana.org/assignments/locally-served-dns-zones/locally-served-dns-zones.xhtml#transport-independent */
+static int dns_special_use_domain_invalid_answer(DnsResourceKey *key, int rcode) {
```

Comment:
if you eat the error from dns_name_endswith() up like you do (and which is ok), then please change the return value here to "bool"